File: domain/src/main/java/de/x4fyr/timeline/domain/elements/TodoElement.java
package de.x4fyr.timeline.domain.elements;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.Duration;
import java.time.LocalDate;

/**
 * Element stashed as a To-Do with optional date and duration.
 *
 * @author Benedikt Volkmer
 *         Created on 11/4/16.
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class TodoElement extends Element {

    private LocalDate plannedDate;
    private Duration plannedDuration;

    public TodoElement(String title, String notes) {
        super(title, notes);
    }

}

File: domain/src/main/kotlin/de/x4fyr/timeline/domain/elements/TodoElement.kt
package de.x4fyr.timeline.domain.elements

import org.joda.time.Duration
import org.joda.time.LocalDate

/**
 * Element stashed as a To-Do with optional date and duration.
 * @author Benedikt Volkmer
 *         Created on 11/4/16.
 */
data class TodoElement(val id: Long? = null,
                       var title: String,
                       var notes: String,
                       var plannedDate: LocalDate? = null,
                       var plannedDuration: Duration? = null)
